engine/rpcserver.go

```diff
@@ -1600,55 +1600,65 @@ func (s *RPCServer) GetAuditEvent(_ context.Context, r *gctrpc.GetAuditEventRequ
 
 // GetHistoricCandles returns historical candles for a given exchange
 func (s *RPCServer) GetHistoricCandles(_ context.Context, req *gctrpc.GetHistoricCandlesRequest) (*gctrpc.GetHistoricCandlesResponse, error) {
-	if req.Exchange == "" {
+	if !req.UseDb && req.Exchange == "" {
 		return nil, errors.New(errExchangeNameUnset)
 	}
 
 	if req.Pair.String() == "" {
 		return nil, errors.New(errCurrencyPairUnset)
 	}
 
-	exchange := GetExchangeByName(req.Exchange)
-	if exchange == nil {
-		return nil, errors.New("Exchange " + req.Exchange + " not found")
-	}
-
 	var candles kline.Item
 	var err error
-	if req.ExRequest {
-		candles, err = exchange.GetHistoricCandlesExtended(currency.Pair{
-			Delimiter: req.Pair.Delimiter,
-			Base:      currency.NewCode(req.Pair.Base),
-			Quote:     currency.NewCode(req.Pair.Quote),
-		},
+	resp := gctrpc.GetHistoricCandlesResponse{
+		Interval: kline.Interval(req.TimeInterval).Short(),
+		Start:    req.Start,
+		End:      req.End,
+	}
+
+	if req.UseDb {
+		candles, err = kline.LoadFromDatabase(req.Exchange,
```

Comment:
Not really a change request just a note. Haven't prefilled any data for a currency pair just gives me back a JSON payload. 

```
{
 "exchange": "poloniex",
 "start": 1595153754,
 "end": 1597832154,
 "interval": "24h"
}
```

Reply:
returns a no data found message now if database query returns no rows